imagescan/controller.go

```diff
@@ -233,6 +237,9 @@ func (s *Controller) scanImages(ctx context.Context, images []*image) error {
 			if err := s.scanImage(ctx, img); err != nil {
 				log.Errorf("image scan failed: %v", err)
 				s.delta.setImageScanError(img, err)
+				if err := s.updateImageScanStatus(ctx, []*image{img}, castai.ImageScanStatusError); err != nil {
```

Comment:
We can add one more status ImageScanStatusPrivate. Check errPrivateImage specific error. This can be useful for us to show user some specific actions to enable private registry.

Reply:
I think we can pass ImageScanStatusError and error_msg with message indicating private registry error